backend/src/procedure/js_scripts.rs

```diff
@@ -605,12 +649,57 @@ mod js_runtime {
             let ctx = state.borrow::<JsContext>();
             Ok(ctx.sandboxed)
         }
+
+        /// We need to make sure that during the file accessing, we don't reach beyond our scope of control
+        fn is_subset(parent: impl AsRef<Path>, child: impl AsRef<Path>) -> bool {
+            let child = normalize_path(child);
+            let parent = normalize_path(parent);
+            child
+                .zip(parent)
+                .map(|(child, parent)| child.starts_with(parent))
+                .unwrap_or(false)
+        }
+
+        /// Used during the subset, we are collapsing the state of the file paths. This will not work with soft link,
+        /// and the correct method, canonicalizePaths seems to freeze randomly, and is much slower.
```

Comment:
This is concerning, we use canonicalize in several places. Evidence?

Reply:
Issue was that the freezing was the order of checking the error from javascript was wrong. Fixed, and made sure the is parent is safer, slower but safer.